Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/TiersRepository.cs

```diff
@@ -53,8 +53,22 @@ public async Task RemoveById(TierId tierId)
         await _tiers.Where(t => t.Id == tierId).ExecuteDeleteAsync();
     }
 
+    public async Task RemoveTierQuotaDefinitionIfOrphaned(TierQuotaDefinitionId tierQuotaDefinitionId)
+    {
+        var tierQuotaDefinition = await _tierQuotaDefinitions.FirstWithId(tierQuotaDefinitionId, CancellationToken.None);
+
+        if (_dbContext.Entry(tierQuotaDefinition).Property("TierId").CurrentValue == null)
+            await _tierQuotaDefinitions.Where(t => t.Id == tierQuotaDefinitionId).ExecuteDeleteAsync();
+    }
+
     public async Task Update(Tier tier, CancellationToken cancellationToken)
     {
+        _dbContext.RemoveRange(_dbContext.ChangeTracker
+            .Entries<TierQuotaDefinition>()
+            .Where(e => e.State == EntityState.Modified)
+            .Select(e => e.Entity)
+            .ToList());
```

Comment:
You should extract this into a private method with a nice name for clarification